org.sonarlint.eclipse.core.tests/src/test/java/org/sonarlint/eclipse/core/internal/engine/AnalysisRequirementNotificationsTest.java

```diff
@@ -68,12 +67,36 @@ public void debug(String msg, boolean fromAnalyzer) {
 
       @Override
       public void showNotification(String title, String shortMsg, String longMsg) {
-        notifications.add(Tuple.tuple(title, shortMsg, longMsg));
+        notifications.add(new Notification(title, shortMsg, longMsg));
       }
 
     });
   }
 
+  private static class Notification {
```

Comment:
My initial idea was to have this type in the prod code, to have more explicit and self-documenting code and group together the 3 properties (title, short and long message). Thiw would mean also changing the API of `showNotification`.